### PR TITLE
Migration tool for ds-1.0.x to ds-2.0.x

### PR DESCRIPTION
**Pull request by NisalaNiroshana**

### CODE REVIEW

File: apps/portal/modules/migration-tool.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    utils.endTenantFlow();
+};
+
+/**
+ * Get the registry path for the id.
+ * @param {String} id Id of the dashboard
+ * @return {String} Registry Path to dashboard
+ * */
+var registryPath = function (id) {
+    var path = '/_system/config/ues/dashboards';
```
Comment (sinthuja) [inline apps/portal/modules/migration-tool.js]: This path variable is not used right?

File: components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+                            + dashboardObj.get(Constants.ID) + Constants.JSON_EXTENSION);
+            filewriter.write(dashboardObj.toJSONString());
+        } catch (IOException e) {
+            log.error("Error in writing dashboard " + dashboardObj.get(Constants.ID) + " into destination path", e);
+        } finally {
+            try {
+                filewriter.flush();
+                filewriter.close();
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (sinthuja) [inline components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java]: Remove print stack trace.

File: components/dashboard-migration-tool/pom.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>carbon-dashboards</artifactId>
+        <groupId>org.wso2.carbon.dashboards</groupId>
+        <version>2.1.1-SNAPSHOT</version>
```
Comment (sinthuja) [inline components/dashboard-migration-tool/pom.xml]: version should be 2.0.1- SNAPSHOT.

File: components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSCarFileMigrationTool.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+                file.write(dashboardJSONObject.toJSONString());
+                log.info("Dashboard " + dashboardJSONObject.get(Constants.ID) + " is successfully updated !");
+            }
+        } catch (ParseException e) {
+            log.error("Error in parsing json file in " + artifactPath, e);
+        } catch (IOException e) {
+            log.error("Error in writing/saving json file in " + artifactPath, e);
+        } finally {
+            try {
+                file.flush();
```
Comment (dineshjweerakkody) [inline components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSCarFileMigrationTool.java]: Check whether file object is null before call flush

File: components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSCarFileMigrationTool.java [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+public class DSCarFileMigrationTool extends DSMigrationTool {
+    private static final Logger log = Logger.getLogger(DSCarFileMigrationTool.class);
+    private static String srcCarDir = null;
+    private static String destCarDir = null;
+    private static String tempDir;
+
+    public static void main(String arg[]) {
+        srcCarDir = arg[0];
+        destCarDir = arg[1];
+        tempDir = srcCarDir.substring(0, srcCarDir.lastIndexOf(File.separator)) + File.separator + Constants.TEMP;
```
Comment (dineshjweerakkody) [inline components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSCarFileMigrationTool.java]: Check the srcCarDir length before call substring. if there is no valid File separator character in the sting lastIndex will be -1 and lead to an exception

File: components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+            file.createNewFile();
+            filewriter = new FileWriter(
+                    destPath + File.separator + Constants.DASHBOARDS + File.separator + tenantDomain + File.separator
+                            + dashboardObj.get(Constants.ID) + Constants.JSON_EXTENSION);
+            filewriter.write(dashboardObj.toJSONString());
+        } catch (IOException e) {
+            log.error("Error in writing dashboard " + dashboardObj.get(Constants.ID) + " into destination path", e);
+        } finally {
+            try {
+                filewriter.flush();
```
Comment (dineshjweerakkody) [inline components/dashboard-migration-tool/src/main/java/org/wso2/carbon/dashboards/migrationtool/DSPortalAppMigrationTool.java]: Check null

**Pull request closed without merging**